tests/tracer/runtime/test_metric_collectors.py

```diff
@@ -28,6 +31,89 @@ def test_metrics(self):
         for (key, value) in collector.collect(PSUTIL_RUNTIME_METRICS):
             self.assertIsNotNone(value)
 
+    def test_static_metrics(self):
+        import os
+        import threading
+        import time
+
+        from ddtrace.vendor import psutil
+
+        # Something to bump CPU utilization
+        def busy_wait(duration_ms):
+            end_time = time.time() + (duration_ms / 1000.0)
+            while time.time() < end_time:
+                pass
+
+        def get_metrics():
+            # need to waste a reading of psutil because some of its reading have
+            # memory and need a previous state
+            collector = PSUtilRuntimeMetricCollector()
+            collector.collect_fn(None)  # wasted
+            proc = psutil.Process(os.getpid())
+            proc.cpu_percent()  # wasted
+
+            # Create some load.  If the duration is too low, then it can cause
+            # wildly different values between readings.
+            busy_wait(50)
+
+            runtime_metrics = dict(collector.collect_fn(None))
+
+            with proc.oneshot():
+                psutil_metrics = {
+                    CPU_PERCENT: proc.cpu_percent(),
+                    MEM_RSS: proc.memory_info().rss,
+                    THREAD_COUNT: proc.num_threads(),
+                }
+            return runtime_metrics, psutil_metrics
+
+        def check_metrics(runtime_metrics, psutil_metrics):
+            def within_threshold(a, b, epsilon):
+                return abs(a - b) <= epsilon * max(abs(a), abs(b))
+
+            # Number of threads should be precise
+            self.assertEqual(psutil_metrics[THREAD_COUNT], runtime_metrics[THREAD_COUNT])
+
+            # CPU and RAM should be approximate.  These tests are checking that the category of
+            # the value is correct, rather than the specific value itself.
+            epsilon = 0.25
+            within_threshold(psutil_metrics[CPU_PERCENT], runtime_metrics[CPU_PERCENT], epsilon)
+            within_threshold(psutil_metrics[MEM_RSS], runtime_metrics[MEM_RSS], epsilon)
+
+        # Sanity-check that the num_threads comparison works
+        rt_metrics, pu_metrics = get_metrics()
+        pu_metrics[THREAD_COUNT] += 1
+        self.assertFalse(check_metrics(rt_metrics, pu_metrics))
```

Comment:
we seem to have an off-by-one error here

Reply:
Ah yeah--I added a negative check (make sure the test fails when it's supposed to fail), but never refactored the way `check_metrics()` returns so that this didn't trigger a fail (because it _should_ fail).  I have a patch, but I forgot to actually push it I guess.

Oops!  Thank you for looking at this carefully, I'm sorry to have wasted your time!